VictoriaAlbert/VictoriaAlbert/RecordTableViewController.swift

```diff
@@ -0,0 +1,169 @@
+//
+//  RecordTableViewController.swift
+//  VictoriaAlbert
+//
+//  Created by Victor Zhong on 11/10/16.
+//  Copyright © 2016 Victor Zhong. All rights reserved.
+//
+
+import UIKit
+
+class RecordTableViewController: UITableViewController, UITextFieldDelegate {
+	
+	
+	@IBOutlet weak var searchField: UITextField!
+	
+	var records = [Record]()
+	var searchTerm = "ring"
+	var locationDict = [String : Int]()
+	var locationArray = [String]()
+	var sorted = false
+	
+	@IBOutlet weak var sortButton: UIBarButtonItem!
+	
+	@IBAction func sortButtonTapped(_ sender: UIBarButtonItem) {
+		sorted = !sorted
+		self.tableView.reloadData()
+		if sorted {
+			sortButton.image = #imageLiteral(resourceName: "filter_filled")
+		}
+		else {
+			sortButton.image = #imageLiteral(resourceName: "filter_empty")
+		}
+	}
+	
+	override func viewDidLoad() {
+		super.viewDidLoad()
+		self.title = searchTerm
+		self.searchField.delegate = self
+		loadRecords(searchString: searchTerm)
+	}
+	
+	internal func loadRecords(searchString: String) {
+		let escapedString = searchTerm.addingPercentEncoding(withAllowedCharacters: .urlHostAllowed)
+		APIRequestManager.manager.getData(endPoint: "http://www.vam.ac.uk/api/json/museumobject/search?q=\(escapedString!)") { (data) in
+			if data != nil {
+				
+				if let records = Record.getRecords(from: data!) {
+					print("We've got Records! \(records.count)")
+					self.records = records
+					self.countLocations()
+					DispatchQueue.main.async {
+						self.tableView.reloadData()
+					}
+				}
+			}
+		}
+	}
+	
+	internal func countLocations() {
+		locationDict = [String : Int]()
+		locationArray = [String]()
+		for all in records {
+			locationDict[all.place] = 0
+		}
+		
+		locationArray = Array(locationDict.keys).sorted { $0 < $1 }
+		print("\n\n\nLocation dict: \(locationDict.count)")
+		print("\nLocation Array: \(locationArray)")
+	}
+	
+	// MARK: - Table view data source
+	
+	override func numberOfSections(in tableView: UITableView) -> Int {
+		if sorted {
+			return locationDict.count
+		}
+		else {
+			return 1
+		}
+	}
+	
+	override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+		if sorted {
+			let recordsByLocation = records.filter { (record) -> Bool in
+				(record.place) == locationArray[section]
+			}
+			
+			return recordsByLocation.count }
+		else {
+			return records.count
+		}
+	}
+	
+	override func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+		guard sorted else { return nil }
+		return "\(locationArray[section])"
+	}
+	
+	override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+		let cell = tableView.dequeueReusableCell(withIdentifier: "recordCell", for: indexPath)
+		var record: Record?
+		if sorted {
```

Comment:
Great extra effort here. One nitpicky thing is that I'd factor out this lookup based on whether the filter/sort is on so you can reuse it in prepare for segue (which at the current moment is duplicated. I know you were just layering stuff on at the time.).